### PR TITLE
Create draft version of AppArmor CRD

### PR DESCRIPTION
**Pull request by pjbgf**
#### What type of PR is this?
/kind feature
/kind api-change

#### What this PR does / why we need it:
Add support for representing AppArmor profiles as CRD.
This reuses #176 and therefore is marked as `Co-authored-by:` @Kaizhe

#### Which issue(s) this PR fixes:
Relates to #118 

#### Does this PR have test?
N/A

#### Special notes for your reviewer:
Reconciler code will be part of a follow-up PR.

The [discussion](https://github.com/kubernetes-sigs/security-profiles-operator/pull/176/files#r545294474) around representing AppArmor profiles in a `structured` vs `unstructured` manner is still up for debate.
Considering this is in `alpha`, the idea was to go ahead implementing the basic behaviour as-is and review it once the functionality is mature enough.

#### Does this PR introduce a user-facing change?

```release-note
Add CRD type to represent AppArmor profiles.
```

### CODE REVIEW

**Comment by pjbgf** [comment]
@Kaizhe thank you for the original contribution, please have a look.

@saschagrunert @cmurphy @JAORMX @jhrozek please share your thoughts

File: examples/apparmorprofile.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+apiVersion: security-profiles-operator.x-k8s.io/v1alpha1
+kind: AppArmorProfile
+metadata:
+  name: apparmorprofile-sample
```
Comment (JAORMX) [inline examples/apparmorprofile.yaml:5]: could you add a simple description of what this does as an annotation here?

**Review by JAORMX — CHANGES_REQUESTED** [review]

**Review by saschagrunert — COMMENTED** [review]
Thanks for taking this over @pjbgf! Happy to lgtm if @JAORMX comment got addressed.

**Comment by pjbgf** [comment]
@saschagrunert @JAORMX changed it accordingly - PTAL

**Review by JAORMX — APPROVED** [review]

**Review by saschagrunert — APPROVED** [review]

**Comment by pjbgf** [comment]
@JAORMX @saschagrunert is the e2e-fedora error is intermittent? is there a way to rerunning it without having to rerun all the git actions tests? 
```
Error:      	Expected nil, but got: command /root/go/src/k8s.io/kubernetes/_output/bin/kubectl -n security-profiles-operator wait --timeout 120s --for condition=ready spod spod did not succeed: error: timed out waiting for the condition on securityprofilesoperatordaemons/spod
```

**Comment by saschagrunert** [comment]
> is there a way to rerunning it without having to rerun all the git actions tests?

No I don't think GitHub actions supports that for now.

**Pull request merged**